### PR TITLE
Feature/Form.TextInput

### PR DESCRIPTION
**Pull request by XDRO**
## Describe your changes
Rebased to main
## Issue ticket number and link
EES-16
https://tripleten-/apiary.atlassian.net/browse/EES-16
## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: src/components/Form/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      <input
+        className={`w-[411.03px] h-[47.04px] p-[12.7px] bg-offwhite rounded-[12px]  ${
+          errors[type] ? "border border-red-500" : ""
+        }`}
+        type={type}
+        placeholder={placeholder}
+        {...register(type, validationRules)}
+        onKeyDown={handleEnter}
+      />
+      {errors[type] && (
```
Comment (RPinkha) [inline src/components/Form/Form.tsx:62]: 2 things about this:
1. The validation message only renders for me when I press enter. Maybe it should render/remove after every keystroke by the user?
2. Need to ask Tomer about this. There seems to be no room for a validation message within the confines of the form. Rendering it to the side of the input will look awkward. Maybe something to mention to our designer?

File: src/components/Form/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        <span className="ftext-red-600 text-sm mt-1">
```
Comment (RPinkha) [inline src/components/Form/Form.tsx]: typo, this needs to be "text-red-600".  Also, for text size please use the ones defined in the config file.

File: src/components/Form/Form.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          ? /^[+]?[0-9\s\-().]{10,15}$/
+          : /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
+      message:
+        type === "tel" ? "Invalid phone number format" : "Invalid email format",
+    },
+  };
+  return (
+    <>
+      <input
+        className={`w-[411.03px] h-[47.04px] p-[12.7px] bg-offwhite rounded-[12px]  ${
```
Comment (RPinkha) [inline src/components/Form/Form.tsx]: I think width of 100% would work better so it takes the width of the container.

**Review by RPinkha — COMMENTED** [review]
Great work. 
Need to update a couple of things, but otherwise looks good to me. 

I forgot to mention. Phone number is actually optional, so maybe do an "or 0" when looking at the valid fields for that input?

**Comment by raymondafan** [comment]
Only thing that needs to be fixed is that the current code only handles the Phone number component. Consider adding a new TextInput component with` type= "email"` to match the jira requirements. Overall, code looks great!
update: looks like i didn't understand the conditional statement for the values, code is good for merge

File: src/components/Form/Form.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          errors[type] ? "border border-red-500" : ""
```
Comment (raymondafan) [inline src/components/Form/Form.tsx:55]: we will discuss the color and positioning of the validation during a zoom call later

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      <Form.TextInput
+        type="tel"
+        placeholder="Phone Number"
+        register={register}
+        errors={errors}
+        handleSubmit={handleSubmit}
+        onSubmit={(data: FieldValues) => {
+          console.log("Submitted data:", data);
+        }}
+      ></Form.TextInput>
```
Comment (raymondafan) [inline src/components/App/App.tsx:23]: Add another Form.Textinput component here for the email
update: looks like i didn't understand the conditional statement for the values, code is good for merge

**Review by raymondafan — APPROVED** [review]
code looks great!

File: src/components/Form/Form.tsx (hunk 3 above)
Comment (XDRO) [reply src/components/Form/Form.tsx]: I understand where you are coming from, but as of now I don't believe it should take the full width of the container. I will bring this up in Tuesday's sync to Tomer though.

File: src/components/Form/Form.tsx (hunk 2 above)
Comment (XDRO) [reply src/components/Form/Form.tsx]: Error message now displayed in red.

File: src/components/Form/Form.tsx (hunk 1 above)
Comment (XDRO) [reply src/components/Form/Form.tsx:62]: I fixed the rendering of the message to be displayed as a column. Line 52. Will make sure to bring up the functionality of the validation to Tomer at Tuesday's sync.

File: src/components/Form/Form.tsx (hunk 2 above)
Comment (XDRO) [reply src/components/Form/Form.tsx]: Corrected text-sm to text-form-text per tailwind.config.js file.

**Pull request closed without merging**